meerk40t/extra/vectrace.py

```diff
@@ -144,3 +123,234 @@ def _vectrace(pixels, width, height):
                     for xi in range(x0, x1):
                         pixels[xi, y0] = 0 if pixels[xi, y0] else 255
                 yield positions
+
+
+def setup_vectrace(kernel):
+    _ = kernel.translation
+
+    @kernel.console_command(
+        "vectrace",
+        help=_("return paths around image"),
+        input_type="image",
+        output_type="elements",
+    )
+    def vectrace(data, **kwargs):
+        elements = kernel.root.elements
+        path = Path(fill="black", stroke="blue")
+        paths = []
+        for node in data:
+            matrix = node.matrix
+            image = node.image
+            width, height = node.image.size
+            if image.mode != "L":
+                image = image.convert("L")
+            image = image.point(lambda e: int(e > 127) * 255)
+            for points in _vectrace(image.load(), width, height):
+                path += Polygon(*points)
+            path.transform *= Matrix(matrix)
+            paths.append(
+                elements.elem_branch.add(
+                    path=abs(path),
+                    stroke_width=0,
+                    stroke_scaled=False,
+                    type="elem path",
+                )
+            )
+        return "elements", paths
+
+"""
+Potracer routines, please be aware that potrace is not part of the standard
+distribution of meerk40t due to the more restrictive licensing of potrace
+If you have installed it yourself via 'pip install potracer', well why not...
+potracer is a pure python port of Peter Selinger's Potrace by Tatarize
+potrace:    https://potrace.sourceforge.net/
+potracer:   https://github.com/tatarize/potrace
```

Comment:
We should probably check if pypotrace works and make sure they are somewhat interchangeable. Pypotrace is potrace compiled from c-source with python bindings. It often crashes the compile and doesn't work consistently. Which is why replacing it with `potracer` would be doable.